docs/proposals/improve-status-in-CAPI-resources.md

```diff
@@ -0,0 +1,1382 @@
+}
+```
+
+| v1beta1 (tentative Dec 2024)      | v1beta2 (tentative Apr 2025)                             | v1beta2 after v1beta1 removal (tentative Apr 2026) |
+|-----------------------------------|----------------------------------------------------------|----------------------------------------------------|
+| `Ready` (deprecated)              | `Ready` (deprecated)                                     | (removed)                                          |
+| `UpdatedReplicas`                 | `UpToDateReplicas` (renamed)                             | `UpToDateReplicas`                                 |
+| `ExperimentalReadyReplicas` (new) | `ReadyReplicas` (renamed)                                | `ReadyReplicas`                                    |
+| `AvailableReplicas` (new)         | `AvailableReplicas`                                      | `AvailableReplicas`                                |
+|                                   | `BackCompatibilty` (new)                                 | (removed)                                          |
+| `ReadyReplicas` (deprecated)      | `BackCompatibilty.ReadyReplicas` (renamed) (deprecated)  | (removed)                                          |
+| `FailureReason` (deprecated)      | `BackCompatibilty.FailureReason` (renamed) (deprecated)  | (removed)                                          |
+| `FailureMessage` (deprecated)     | `BackCompatibilty.FailureMessage` (renamed) (deprecated) | (removed)                                          |
+| `Conditions`                      | `BackCompatibilty.Conditions` (renamed) (deprecated)     | (removed)                                          |
+| `ExperimentalConditions` (new)    | `Conditions` (renamed)                                   | `Conditions`                                       |
+| other fields...                   | other fields...                                          | other fields...                                    |
+
+TODO: double check usages of status.ready.
+
+#### KubeadmControlPlane (New)Conditions
+
+| Condition               | Note                                                                                                                                                                                                                                                                      |
+|-------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `Available`             | True if the control plane can be reached, `EtcdClusterAvailable` is true, and `CertificatesAvailable` is true                                                                                                                                                             |
```

Comment:
does this mean that 'Available' can be true even though 'MachinesAvailable' is false?

why not include 'MachinesAvailable' here?

Reply:
I think because not all Machines have to be available for the control plane to be Available (we have redundancy there)